### PR TITLE
Present more useful exception when  QuarkusTestResourceLifecycleManager fails

### LINKED ISSUE #33371: NPE in QuarkusTestExtension after exception in `QuarkusTestResourceLifecycleManager.start()`
### Describe the bug

By mistake, I had caused an exception in the `start` method of my custom `MariaDBContainerQuarkusLifecycleManager`.

That exception is printed just as expected, but after some skipped QuarkusTests the following exception pops up:

```
java.lang.NullPointerException: Cannot invoke "io.quarkus.bootstrap.app.RunningQuarkusApplication.getClassLoader()" because "io.quarkus.test.junit.QuarkusTestExtension.runningQuarkusApplication" is null
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:901)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:893)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptAfterAllMethod(QuarkusTestExtension.java:887)
        [...]
	Suppressed: org.opentest4j.TestAbortedException: Boot failed
		at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:627)
		at io.quarkus.test.junit.QuarkusTestExtension.interceptBeforeAllMethod(QuarkusTestExtension.java:680)
		at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
```
That NPE is a bit misleading and shouldn't be there.

### Expected behavior

No NPE

### Actual behavior

NPE as described above

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.7.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #33371

P.S. I didn't see the same stacktrace as #33371, but that's probably because that is using Quarkus 2.x

### CODE REVIEW

File: test-framework/derby/src/main/java/io/quarkus/test/derby/DerbyDatabaseTestResource.java [hunk 1]
```diff
@@ -15,31 +13,33 @@ public class DerbyDatabaseTestResource implements QuarkusTestResourceLifecycleMa
...
+        //                        Thread.sleep(SLEEP_BETWEEN_PINGS);
+        //                    } catch (InterruptedException ignore) {
+        //                    }
+        //                }
+        //            }
+        //            System.out.println("[INFO] Derby database started in TCP server mode");
+        //        } catch (Exception e) {
+        //            throw new RuntimeException(e);
+        //        }
+        //        return Collections.emptyMap();
```
Comment (Sgitario) [inline test-framework/derby/src/main/java/io/quarkus/test/derby/DerbyDatabaseTestResource.java:42]: I think this change is not intentional, right?

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: test-framework/derby/src/main/java/io/quarkus/test/derby/DerbyDatabaseTestResource.java (hunk 1 above)
Comment (geoand) [reply test-framework/derby/src/main/java/io/quarkus/test/derby/DerbyDatabaseTestResource.java:42]: Oops!

**Review by Sgitario — APPROVED** [review]

**Comment by famod** [comment]
I'm a bit confused about how this is supposed to fix that NPE I reported in #33371? Are we assuming it's not thrown in Q3?

PS: I would love to test-drive this PR, but blaze-persistence prevents me from upgrading. I could probably backport it locally... 🤔

**Comment by geoand** [comment]
@famod I did the same thing you did but never got the NPE, so yeah, I am assuming it doesn't happen anymore with Quarkus 3

**Pull request merged**